### PR TITLE
fix: migrate from `node-sass` to `dart-sass`

### PR DESCRIPTION
**Pull request by harish-sethuraman**
Screenshot 2022-05-22 at 12 48 32 PM

`node-sass` is deprecated also it throws errors when the engines are not compatible when using latest node versions. `dart-sass` is better

Also: 

- Added css compression to compress the generated CSS assets
- added `--no-source-map` to avoid sourcemap generation

### CODE REVIEW

**Comment by harish-sethuraman** [comment]
Screenshot 2022-05-23 at 3 44 54 PM
We might need if we want to show the exact line number in the inspect tool? But I'm not sure didn't think it would be useful so removed it. May be we can enable it in the dev alone?

File: package.json [hunk 1]
```diff
@@ -9,10 +9,10 @@
     "scripts": {
         "images": "imagemin '_site/assets/images' --out-dir='_site/assets/images'",
         "install:playground": "cd src/playground && npm install --no-package-lock",
-        "watch:sass": "node-sass --watch src/assets/scss -o src/assets/css",
+        "watch:sass": "sass --style=compressed --no-source-map --watch src/assets/scss:src/assets/css",
```
Comment (aladdin-add) [inline package.json:12]: or disable compressing for faster dev?

Comment (harish-sethuraman) [inline package.json:12]:
```suggestion
        "watch:sass": "sass --watch src/assets/scss:src/assets/css",
```

or we can do both :)

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks!

**Comment by harish-sethuraman** [comment]
Bumping for merge in case this was missed

**Review by snitin315 — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**